src/adapters/adapter.js

```diff
@@ -167,7 +167,7 @@ class Adapter {
         } else {
           error = 'Forbidden';
           message =
-            'Accessing private repositories requires a GitHub access token. ' +
+            'A GitHub access token is required to access private repositories, or when the API rate limit is exceeded. ' +
```

Comment:
You know when this case is activated? I mean I thought private repo should hit the 'case 404' and api limit should hit the first if block of 'case 403'. If those are not handled correctly, should we update them?

Reply:
This check is activated in these cases : 
- GH rate limit exceeded and server responds with 403.
- GH blocks a certain IP and server responds with 403.

So yes there's a bug and it's due to the check at line 160 is wrong ```jqXHR.getResponseHeader('X-Ratelimit-Remaining')``` 
this line returns null when rate limits is exceeded. The function is returning null maybe (due to some issue with CROS).

If can't find a way to read the rate limit header for now first thought i have in mind is to currently go with the suggestion described in the issue and keep a generic error message for 403 case like this : 

```A GitHub access token is required to access private repositories, or when the API rate limit is exceeded.```

Reply:
Can we detect the response to fix the root cause instead? So that it shows the full error message in line 164 (which explains what rate limit is).